### PR TITLE
add serial to upper option to prompt_for_test_start

### PR DESCRIPTION
**Pull request by wallacbe**
https://github.com/google/openhtf/issues/485

### CODE REVIEW

**Comment by wallacbe** [comment]
PTAL

File: openhtf/plugs/user_input.py [hunk 1]
```diff
@@ -194,13 +194,15 @@ def respond(self, prompt_id, response):
 
 
 def prompt_for_test_start(
-    message='Enter a DUT ID in order to start the test.', timeout_s=60*60*24):
+    message='Enter a DUT ID in order to start the test.', timeout_s=60*60*24,
+    upper=False):
```
Comment (fahhem) [inline openhtf/plugs/user_input.py]: I'd prefer a validation hook that's a little more general and you can provide `string.upper` but in the future you may want to validate serials more thoroughly.

Take `validator=None` as an argument and put this below:

```py
if validator is None:
  validator = lambda sn: sn
test.test_record.dut_id = validator(dut_id)
```

**Comment by jettisonjoe** [comment]
My thinking on this is that it seems a bit random to add an 'upper' feature to just this start trigger. Mainly it seems like a project-specific need, and `prompt_for_test_start` is already a _very_ thin wrapper around the prompt mechanism to begin with. Some alternate approaches I think would be better:

1. Define your own start trigger that wraps prompting just like `prompt_for_test_start` does, but `upper`'s the text (as if your `upper` arg here were true).
2. Use `prompt_for_test_start` as-is, but reach into the test record later and upper the `dut_id` by hand.
3. Instead of a specific `upper` argument, add a `format_callback`. If the arg is `None` (the default), the start trigger does what it currently does. Otherwise, the start trigger sets `dut_id` to the result of `format_callback(raw_dut_id)`. I can also imagine variations on this where `format_callback` is set as an attribute of the start trigger.
4. Some pipe-style or other more general dut_id formatting approach even more clever than 3 that I haven't thought of.

**Comment by wallacbe** [comment]
PTAL

**Comment by wallacbe** [comment]
Done, added docstring

**Comment by wallacbe** [comment]
Thanks, PTAL maintainer

**Comment by fahhem** [comment]
LGTM

**Comment by fahhem** [comment]
Oops, I forgot to rename the commit!

**Pull request merged**